File: src/main/java/com/mrozowski/elasticsearchdemo/domain/model/Product.java
package com.mrozowski.elasticsearchdemo.domain.model;

import lombok.Builder;

@Builder
public record Product(String id, String name, String description, Category category, int quantity, int priceInCents) {


  static public enum Category {
    ELECTRONICS,
    APPLIANCES,
    CLOTHES
  }
}

File: src/main/java/com/mrozowski/elasticsearchdemo/domain/model/Product.kt
package com.mrozowski.elasticsearchdemo.domain.model

data class Product(
    val id: String? = null,
    val name: String,
    val description: String,
    val category: Category,
    val quantity: Int,
    val priceInCents: Int
) {
    enum class Category {
        ELECTRONICS,
        APPLIANCES,
        CLOTHES
    }
}
